### ISSUE TITLE
Julia 1.0.0 GLPKOptimizerMIP() not defined

### ISSUE THREAD

**Issue by monetschemist** [JuliaOpt/GLPK.jl#77]
First steps with Julia, starting with Julia 1.0 and trying to follow the knapsack problem example here

http://www.juliaopt.org/MathOptInterface.jl/latest/apimanual.html#A-complete-example:-solving-a-knapsack-problem-1

My attempts to make this work fail as follows:

```
$ julia
               _
   _       _ _(_)_     |  Documentation: https://docs.julialang.org
  (_)     | (_) (_)    |
   _ _   _| |_  __ _   |  Type "?" for help, "]?" for Pkg help.
  | | | | | | |/ _` |  |
  | | |_| | | | (_| |  |  Version 1.0.0 (2018-08-08)
 _/ |\__'_|_|_|\__'_|  |  Official https://julialang.org/ release
|__/                   |

julia> using MathOptInterface

julia> const MOI = MathOptInterface
MathOptInterface

julia> using GLPK

julia> optimizer = GLPK.GLPKOptimizerMIP()
ERROR: UndefVarError: GLPKOptimizerMIP not defined
Stacktrace:
 [1] getproperty(::Module, ::Symbol) at ./sysimg.jl:13
 [2] top-level scope at none:0
```
I see in this repository that GLPK is tested against Julia 0.6 and 0.7.  Perhaps it is not yet meant to work against 1.0.0?

Thanks in advance.

**Comment by rdeits** [comment]
Ah, it looks like there are two issues: 

1. That example hasn't been updated for the recent changes to GLPK. You want `GLPK.Optimizer()` instead of `GLPK.GLPKOptimizerMIP()`
2. GLPK itself hasn't been updated to be compatible with the latest MathOptInterface, so you'll want to follow the example from the MathOptInterface v0.5.1 version of the manual: http://www.juliaopt.org/MathOptInterface.jl/release-0.5/apimanual.html#A-complete-example:-solving-a-knapsack-problem-1

**Comment by rdeits** [comment]
Actually, my point (2) wasn't quite right: GLPK.jl *has* been updated, but there wasn't yet a released tag. You can either follow the older instructions as I suggested, or you can tell Julia to use the `master` branch of this package by running: 

```
pkg> add GLPK#master
```

where you can press `]` to switch the `julia>` prompt into `pkg>` mode.

**Comment by monetschemist** [comment]
Thank you very much for this info! I will give it a try before the weekend
is out and report back. I really appreciate the assistance.

Chris Hermansen

**Comment by monetschemist** [comment]
I realize this issue is closed but I thought it might be worthwhile to report my results.

I took the approach of using the package manager to add the package GLPK#master as suggested.  Then I followed the version of the program documented [http://www.juliaopt.org/MathOptInterface.jl/latest/apimanual.html#A-complete-example:-solving-a-knapsack-problem-1](url).

This worked without a hitch.  I also substituted in a problem (with 216 items) for which I had used a dynamic programming approach in Groovy, based on the example on Rosetta Code here [https://rosettacode.org/wiki/Knapsack_problem/0-1#Groovy](url).  Both versions arrived at the same solution and objective value and the Julia version took about 1/3 the time - 6 seconds in Julia vs. 19 in Groovy - on an older i7 laptop with 8Gb of memory.

In summary, all good from my end.  Thanks very much for the great support.

**Comment by rdeits** [comment]
That's great to hear! Thanks for reporting back.
